src/neuroconv/datainterfaces/ecephys/maxwell/maxonedatainterface.py

```diff
@@ -0,0 +1,82 @@
+"""Authors: Cody Baker."""
+from pathlib import Path
+from platform import system
+from typing import Optional
+
+from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
+from ....utils.types import FilePathType, FolderPathType
+
+
+class MaxOneRecordingInterface(BaseRecordingExtractorInterface):
+    """
+    Primary data interface class for converting MaxOne data.
+
+    Using the :py:class:`~spikeinterface.extractors.MaxwellRecordingExtractor`.
+    """
+
+    ExtractorName = "MaxwellRecordingExtractor"
+
+    @staticmethod
+    def auto_install_maxwell_hdf5_compression_plugin(
+        hdf5_plugin_path: Optional[FolderPathType] = None, download_plugin: bool = True
+    ) -> None:
+        """
+        If you do not yet have the Maxwell compression plugin installed, this function will automatically install it.
+
+        Parameters
+        ----------
+        hdf5_plugin_path : string or Path, optional
+            Path to your systems HDF5 plugin library.
+            Uses the home directory by default.
+        download_plugin : boolean, default: True
+            Whether or not to force download of the decompression plugin.
+            It's a very lightweight install but does require an internet connection.
+            This is left as True for seamless passive usage and should not impact performance.
```

Comment:
Note that `neo` takes the same approach by default, for conveneience: https://github.com/NeuralEnsemble/python-neo/blob/master/neo/rawio/maxwellrawio.py#L229

If you swap the convention, then it introduces a workflow the user has to go through where they need to manage the extra installation and setup before instantiating the interface

Reply:
But there might be cases where the user wants or needs that control, hence why I'm including this as an argument where informed users may disable it by setting it to `False` and managing it the way they want